### PR TITLE
[Version] Increment main to 2.0

### PR DESCRIPTION
**Pull request by nknize**
Add next version 2.0.

depends #1191

### CODE REVIEW

**Comment by nknize** [comment]
bwc test w/ 1.0.0 and 1.0.1 will fail until #1191 is merged

File: test/framework/src/test/java/org/opensearch/test/VersionUtilsTests.java [hunk 1]
```diff
@@ -224,12 +224,12 @@ public void testResolveReleasedVersionsAtNewMajorRelease() {
     }
 
     public static class TestVersionBumpIn2x {
-        public static final Version V_1_6_0 = Version.fromString("1.6.0");
-        public static final Version V_1_6_1 = Version.fromString("1.6.1");
-        public static final Version V_1_6_2 = Version.fromString("1.6.2");
-        public static final Version V_2_0_0 = Version.fromString("2.0.0");
-        public static final Version V_2_0_1 = Version.fromString("2.0.1");
-        public static final Version V_2_1_0 = Version.fromString("2.1.0");
+        public static final Version V_1_6_0 = Version.fromString("2.6.0");
```
Comment (nknize) [inline test/framework/src/test/java/org/opensearch/test/VersionUtilsTests.java:227]: @todo change statics to reflect versions

**Comment by nknize** [comment]
start gradle check

**Comment by nknize** [comment]
start gradle check

**Comment by nknize** [comment]
start gradle check

**Comment by nknize** [comment]
start gradle check

**Review by dblock — APPROVED** [review]

**Pull request merged**